sr_world_generator/src/sr_world_generator/clean_world_file.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+import rospy
+
+
+class WorldFileCleaner:
+    """
+    This class is used to clean up a Gazebo world file by removing named models and state information.
+    This is useful when you want to save a world file that has been modified by Gazebo, but you don't want to save the
+    state information or any models (e.g. robots) that you don't want to be in the world file.
+    """
+    _xml_tree: ET.ElementTree = None
+
+    def __init__(self, dry_run: bool = False):
+        self._dry_run = dry_run
+
+    def load_world_file(self, file_path: str):
+        """
+        Loads the world file from the specified path.
+
+        Args:
+            file_path: The path to the world file to load.
+        """
+        rospy.loginfo(f'Loading world file from {file_path}')
+        # Check if the file exists
+        if not os.path.isfile(file_path):
+            rospy.logerr(f'World file does not exist: {file_path}')
+            return False
```

Comment:
I thought using multiple returns for flow control was frowned on?

Reply:
Not that I know of. Why would it be? Not sure what the alternative is.